controllers/pipelines/pipeline_controller.go

```diff
@@ -56,7 +58,16 @@ func (r *PipelineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	desiredProvider := desiredProvider(pipeline, r.Config)
 
-	commands := r.StateHandler.StateTransition(ctx, desiredProvider, pipeline)
+	provider := pipelinesv1.Provider{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: desiredProvider,
+		},
+		Spec:   pipelinesv1.ProviderSpec{},
+		Status: pipelinesv1.ProviderStatus{},
```

Comment:
This creates an empty provider object and never reads it from the cluster. Have we missed that?

Reply:
This is intended for this card, this is using the provider struct but for the moment only to carry the desiredProvider string name to the workflow. 

The next step is to actually load in the provider and populate these fields correctly